src/Swarm/TUI/Model/Achievement/Persistence.hs

```diff
@@ -4,38 +4,73 @@ module Swarm.TUI.Model.Achievement.Persistence where
 
 import Control.Arrow (left)
 import Control.Carrier.Lift (sendIO)
+import Control.Monad (forM, forM_)
+import Data.Either (partitionEithers)
 import Data.Text (Text, pack)
+import Data.Text qualified as T
 import Data.Yaml qualified as Y
 import Data.Yaml.Aeson (prettyPrintParseException)
 import Swarm.TUI.Model.Achievement.Attainment
 import Swarm.Util
 import System.Directory (
+  createDirectoryIfMissing,
+  doesDirectoryExist,
   doesFileExist,
+  listDirectory,
  )
 import System.FilePath
+import System.IO (hPutStrLn, stderr)
 
 -- | Get path to swarm achievements, optionally creating necessary
 --   directories.
 getSwarmAchievementsPath :: Bool -> IO FilePath
-getSwarmAchievementsPath createDirs =
-  (</> "achievements") <$> getSwarmDataPath createDirs
+getSwarmAchievementsPath createDirs = do
+  dp <- getSwarmDataPath False
+  let achievementsDir = dp </> "achievement"
+  createDirectoryIfMissing createDirs achievementsDir
+  return achievementsDir
 
 -- | Load saved info about achievements from XDG data directory.
 loadAchievementsInfo ::
   IO (Either Text [Attainment])
 loadAchievementsInfo = do
-  savedAchievementsPath <- getSwarmAchievementsPath True
-  hasSavedAchievements <- sendIO $ doesFileExist savedAchievementsPath
-  if not hasSavedAchievements
-    then return $ pure []
-    else
-      left (pack . prettyPrintParseException)
-        <$> sendIO (Y.decodeFileEither savedAchievementsPath)
+  savedAchievementsPath <- getSwarmAchievementsPath False
+  doesParentExist <- doesDirectoryExist savedAchievementsPath
+  if doesParentExist
+    then do
+      contents <- listDirectory savedAchievementsPath
+      eithersList <- forM contents $ \p -> do
+        let fullPath = savedAchievementsPath </> p
+        isFile <- doesFileExist fullPath
+        if isFile
+          then do
+            eitherDecodedFile <- sendIO (Y.decodeFileEither fullPath)
+            return $ left (pack . prettyPrintParseException) eitherDecodedFile
+          else
+            return $
+              Left $
+                T.unwords
+                  [ "Entry"
+                  , quote $ T.pack fullPath
+                  , "is not a file!"
+                  ]
+      let (bads, goods) = partitionEithers eithersList
+      forM_ bads $ \b ->
+        hPutStrLn stderr $ T.unpack b
```

Comment:
Please don’t put stuff to stderr.

We have a RuntimeState with logs for this purpose.

Let me describe why:
1. I run the game
2. an error is printed shortly before Brick opens
3. I play Swarm puzzled why some things are wrong
4. I close the game and see the log
5. I wonder why I am only seeing this now and not in game notifications

Reply:
Sure, let's log using the existing mechanism.  Could you please illustrate how?  I've pushed a commit atop this branch with an incomplete attempt.